eos-install-firefox

```diff
@@ -0,0 +1,358 @@
+JAVA_FILENAME=jre-${JAVA_VERSION}-linux-x64.tar.gz
+JAVA_URL_FILENAME=http://download.oracle.com/otn-pub/java/jdk/${JAVA_BUILD}/${JAVA_FILENAME}
+
+# While the following would be a faster download of the current version,
+# previous versions are often removed from the server, so it is safest
+# for an automated script to use the stable version in the archive
+# http://fpdownload.macromedia.com/get/flashplayer/pdc/11.2.202.643/install_flash_player_11_linux.x86_64.tar.gz
+FLASH_VERSION=11.2.202.643
+FLASH_SHA256SUM="1a79f38f7153d436083b5a6b5d0f38e0c4cd345e83e7d050152452057f83350a"
+FLASH_FILENAME=fp_${FLASH_VERSION}_archive.zip
+FLASH_URL_FILENAME=http://download.macromedia.com/pub/flashplayer/installers/archive/${FLASH_FILENAME}
+
+ICON_FILENAME=firefox.png
+ICON_URL_FILENAME=https://raw.githubusercontent.com/endlessm/eos-customer-support/master/data/${ICON_FILENAME}
+
+get_firefox_lang() {
+    # List of supported Firefox languages, from https://download-installer.cdn.mozilla.net/pub/firefox/releases/latest/README.txt
+    FIREFOX_LANGS="ach af an ar as ast az be bg bn-BD bn-IN br bs ca cs cy da de dsb el en-GB en-US en-ZA eo es-AR es-CL es-ES es-MX et eu fa ff fi fr fy-NL ga-IE gd gl gu-IN he hi-IN hr hsb hu hy-AM id is it kk km kn ko lij lt lv mai mk ml mr ms nb-NO nl nn-NO or pa-IN pl pt-BR pt-PT rm ro ru si sk sl son sq sr sv-SE ta te th tr uk uz vi xh zh-CN zh-TW"
+
+    # Find the closest supported Firefox for the user's current language
+    locale=`echo ${LANG} | cut -f 1 -d . | sed s/_/-/`
+    lang=`echo ${locale} | cut -f 1 -d -`
+    for l in ${FIREFOX_LANGS}
+    do
+	if [ ${l} == ${locale} ]
+	then
+            FIREFOX_LANG=${l}
+            break
+	fi
+    done
+
+    if [[ -z ${FIREFOX_LANG} ]]; then
+	for l in ${FIREFOX_LANGS}
+	do
+	    if [ ${l} == ${lang} ]
+	    then
+		FIREFOX_LANG=${l}
+		break
+	    fi
+	done
+    fi
+
+    if [[ -z ${FIREFOX_LANG} ]]; then
+	if [ ${lang} == es ]; then
+	    FIREFOX_LANG=es-MX
+	else
+	    FIREFOX_LANG=en-US
+	fi
+    fi	
+}
+
+download_packages() {
+    echo "Downloading ${FIREFOX_FILENAME}"
+    FIREFOX_LOCALIZED_URL="${FIREFOX_URL}&lang=${FIREFOX_LANG}"
+    if ! ${WGET_COMMAND} "${FIREFOX_LOCALIZED_URL}" -O "${TEMP_DIR}/${FIREFOX_FILENAME}"; then
+        exit_with_error "Failed to download ${FIREFOX_LOCALIZED_URL}"
+    fi
+
+    echo "Downloading ${JAVA_FILENAME}"
+    if ! ${WGET_COMMAND} --no-cookies --no-check-certificate --header "Cookie: oraclelicense=accept-securebackup-cookie" ${JAVA_URL_FILENAME} -O "${TEMP_DIR}/${JAVA_FILENAME}"; then
+        exit_with_error "Failed to download ${JAVA_URL_FILENAME}"
+    fi
+
+    echo "Downloading ${FLASH_FILENAME}"
+    if ! ${WGET_COMMAND} "${FLASH_URL_FILENAME}" -O "${TEMP_DIR}/${FLASH_FILENAME}"; then
+	exit_with_error "Failed to download ${FLASH_URL_FILENAME}"
+    fi
+
+    echo "Verifying ${TEMP_DIR}/${FLASH_FILENAME}"
+    echo "${FLASH_SHA256SUM} ${TEMP_DIR}/${FLASH_FILENAME}" | sha256sum -c > /dev/null 2>&1 || \
+    {
```

Comment:
And, actually, I think it is safest to keep them in case anyone tries to add another statement, since there is no "if" / "fi" or "do" / "done" delimiters.